src/fem/fe_values.hh

```diff
@@ -256,9 +263,10 @@ public:
      * @param _fe The finite element.
      * @param flags The update flags.
      */
-    void allocate(Mapping<dim,spacedim> &_mapping,
+    virtual void allocate(Mapping<dim,spacedim> &_mapping,
```

Comment:
Why making this virtual, if it is called only from constructor which is separate for FEM and XFEM case.